src/chart/heatmap/HeatmapView.ts

```diff
@@ -203,8 +206,8 @@ class HeatmapView extends ChartView {
             let rect;
 
             if (isCoordinateSystemType<Cartesian2D>(coordSys, 'cartesian2d')) {
-                // Ignore empty data
-                if (isNaN(data.get(dataDims[2], idx) as number)) {
+                // Ignore empty data and out of extent data
+                if (isNaN(data.get(dataDims[2], idx) as number) || data.get(dataDims[0], idx) < xAxisExtent[0] || data.get(dataDims[0], idx) > xAxisExtent[1]) {
                     continue;
```

Comment:
data.get(dataDims[0], idx) is better to be assigned to a temporal variable.

Also, yAxis should also be checked?

Reply:
I agree, have modified accordingly.